proposals/0000-style-script-defaults.md

```diff
@@ -0,0 +1,88 @@
+- Start Date: 2022-03-15
+- Reference Issues: [RFC0008](https://github.com/withastro/rfcs/tree/main/proposals/0008-style-script-behavior.md), [#65](https://github.com/withastro/rfcs/discussions/65)
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+Astro is inconsist between `<style>` and `<script>` default behavior. Currently, `<style>` has opt-out processing, but `<script>` does not. This RFC aims to settle on a good, consistent default for both `<style>` and `<script>`.
+
+- New `is:inline` directive to avoid bundling `style` or `script`
+- `<style>` => remains scoped, bundled by default. `is:scoped` directive supported for consistency
+- `<style global>` => becomes `<style is:global>`
+- `<script>` => becomes `<script is:inline>`
+- `<script hoist>` => removed, becomes default `<script>` behavior. `type="module"` is implied
+
+# Example
+
+```astro
+<style>
+  @import './dep.css'; /* Vite imports supported, including npm packages! */
+  h1 { color: red; }
+</style>
+
+<script>
+  import cowsay from 'cowsay';
+  console.log(cowsay('I am bundled with the rest of the website JS!'));
+</script>
+
+<h1>Hello!</h1>
+
+<script is:inline>
+console.log("I am literally inlined here on the page");
+</script>
+<style is:inline>
+span { color: green; }
+</style>
+```
+
+# Motivation
+
+There have been [a few](https://github.com/withastro/rfcs/pull/12) [attempts](https://github.com/withastro/rfcs/discussions/65) to finalize this behavior and address these inconsistent APIs. This RFC aims to settle these questions prior to v1.0.
+
+- Users are generally in favor of how we currently do scoped, processed, and optimized styles by default. It is logical to extend this feature to scripts.
+- Inlining scripts and styles can be useful for anything that must exist inlined into the page, like Google Analytics. Users currently cannot do this for styles.
+- Users currently have to remember syntax for styles and scripts that is unlike other Astro directives.
+- Encourages our "optimized by default" / "pit of success" mentality. Smart scripts should be easy to use by default.
+
+# Detailed design
+
+### `is:inline`
+
+A new directive, `is:inline`, will be introduced. This will opt both `<style>` and `<script>` tags out of bundling behavior.
+
+### `is:scoped`, `is:global`
+
+These new directives are for `<style>` tags. They represent scoped (the default) and global (previously `global`) styles.
+
+### `hoist` by default
+
+Support for `<script hoist>` will be removed. This will become the default `<script>` behavior, where `type="module"` is implied and scripts are preprocessed and bundled by default.
+
+If any script tag has an attribute (ex: `src=`, `type="module"`, `async`) or directive (`define:vars`), it is implied to be `is:inline`. Astro should emit a warning that `is:inline` is needed for clarity.
```

Comment:
The `src=` caveat here is slightly different from the current hoist behavior in that you can have external hoisted scripts. I don't think there's a technical reason to have this restriction. Is there another reason?

For clarity, you could still do this:

```js
<script>
  import "../scripts/external.js";
</script>
```

I think this is ok and aligns with how `style` currently works.

Reply:
Updated to remove `src`!